tests_v2/test_read_selection.py

```
@@ -0,0 +1,147 @@
+        surfaces = surfaces_payload.get("surfaces") or []
+        _must(bool(surfaces), f"Expected at least one surface in workspace: {surfaces_payload}")
+        sf = str(surfaces[0].get("id") or "")
+        _must(bool(sf), f"surface.list returned surface without id: {surfaces_payload}")
+
+        # 1) Fresh terminal: no active selection.
+        payload = c._call("surface.read_selection", {"workspace_id": ws, "surface_id": sf}) or {}
+        _must(payload.get("has_selection") is False, f"Expected has_selection false on fresh terminal: {payload}")
+        _must(str(payload.get("text") or "") == "", f"Expected empty text without selection: {payload}")
+
+        # 2) CLI without selection: non-zero exit + explicit error.
+        no_sel = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)